spec/rails_helper.rb

```diff
@@ -1,6 +1,14 @@
 # frozen_string_literal: true
 
 ENV['RAILS_ENV'] ||= 'test'
+
+# This needs to be defined before Rails is initialized
+RUN_SYSTEM_SPECS = ENV.fetch('RUN_SYSTEM_SPECS', false)
+
+if RUN_SYSTEM_SPECS
+  STREAMING_PORT = rand(4001..4999)
```

Comment:
It seems sort of odd to have a random assignment like this for an env that in theory we fully control (the CI setup) ... whats the background here?

Reply:
Change it an a specific env variable, with a default.

The main issue was that re-using the port 4000 was conflicting with any local dev environment you might be running, so I figured that making it random would solve this and allow parallel system tests if needed as some point.

But it makes more sense to have it set by the environment, so here we go